aospy/automate.py

```diff
@@ -0,0 +1,220 @@
+                                               'runs', 'default_runs')
+                for run in runs:
+                    # Use Calc's attr names as keys.
+                    obj_trees.append(dict(proj=project, model=model, run=run))
+        return obj_trees
+
+    def _permute_aux_specs(self):
+        # Drop the "core" specifications, which are handled separately.
+        specs = self._specs_in.copy()
+        [specs.pop(core) for core in self._CORE_SPEC_NAMES]
+        # TODO: don't just pull from first project.
+        regions = _get_mult_objs_by_names(specs['regions'], self._projects[0],
+                                           'regions', 'regions')
+        variables = _get_mult_objs_by_names(specs['variables'], self._obj_lib,
+                                            'variables', 'variables')
+        specs['regions'] = regions
+        specs['variables'] = variables
+
+        # Convert to attr names that Calc is expecting.
+        calc_aux_mapping = self._NAMES_SUITE_TO_CALC_MAPPING.copy()
+        [calc_aux_mapping.pop(core) for core in self._CORE_SPEC_NAMES]
+        for suite_name, calc_name in calc_aux_mapping.items():
+            specs[calc_name] = specs.pop(suite_name)
+        return _permute_aux_specs(specs)
+
+    def _combine_core_aux_specs(self):
+        return _combine_core_aux_specs(self._permute_core_specs(),
+                                       self._permute_aux_specs())
+
+    def create_calcs(self):
+        return _create_calcs(self._combine_core_aux_specs())
+
+    def _print_specs(self):
+        pairs = (
+            ('Projects', self.projects),
+            ('Models', self.models),
+            ('Runs', self.runs),
+            ('Variables', self.variables),
+            ('Date ranges', self.date_ranges),
+            ('Geographical regions', [r.values() for r in self.region]),
+            ('Time interval of input data', self.intvl_in),
+            ('Time interval for averaging', self.intvl_out),
+            ('Input data time type', self.dtype_in_time),
+            ('Input data vertical type', self.dtype_in_vert),
+            ('Output data time type', self.dtype_out_time),
+            ('Output data vertical type', self.dtype_out_vert),
+            ('Time offset', self.time_offset),
+        )
+        print('')
+        try:
+            colorama.init()
+            color_left = colorama.Fore.BLUE
+            color_right = colorama.Fore.RESET
+            reset = colorama.Style.RESET_ALL
+        except NameError:
+            color_left = ''
+            color_right = ''
+            reset = ''
+        for left, right in pairs:
+            print(color_left, left, ':', color_right, right)
+        print(reset)
+
+
+def exec_calcs(calcs, parallelize=False):
+    if parallelize:
+        pool = multiprocess.Pool()
+        return pool.map(lambda calc: calc.compute(), calcs)
+    out = []
+    for calc in calcs:
+        try:
```

Comment:
Could there be a way to encapsulate this try-except logic such that it could also be used above when we parallelize the calculations?  That way it would ensure the same warning behavior through each pathway.